zipline/gens/tradesimulation.py

```diff
@@ -216,15 +214,19 @@ def calculate_minute_capital_changes(dt):
             for dt, action in self.clock:
                 if action == BAR:
                     every_bar(dt)
-                elif action == DAY_START:
+                elif action == SESSION_START:
                     once_a_day(dt)
-                elif action == DAY_END:
-                    # End of the day.
-                    if emission_rate == 'daily':
+                elif action == SESSION_END:
+                    # End of the session.
+                    if algo.perf_tracker.emission_rate == 'daily':
                         handle_benchmark(normalize_date(dt))
                     execute_order_cancellation_policy()
 
                     yield self._get_daily_message(dt, algo, algo.perf_tracker)
+                elif action == BEFORE_TRADING_START_BAR:
+                    # call before trading start
+                    algo.on_dt_changed(dt)
```

Comment:
If the sim engine might emit bars within a session both before and after BTS, might the dt not change here?